apps/api/v2/src/modules/workflows/outputs/workflow.output.ts

```diff
@@ -0,0 +1,274 @@
+import { ApiProperty, ApiPropertyOptional } from "@nestjs/swagger";
+import { Expose, Type } from "class-transformer";
+import { IsArray, IsEnum, ValidateNested } from "class-validator";
+
+// Assuming these constants are available
+const SUCCESS_STATUS = "success" as const;
+const ERROR_STATUS = "error" as const;
+type Status = typeof SUCCESS_STATUS | typeof ERROR_STATUS;
+
+// --- Re-using Enums from your Input DTOs ---
+// (Alternatively, import them if they are in separate files)
+export enum WorkflowTriggerType {
+  BEFORE_EVENT = "BEFORE_EVENT",
+  EVENT_CANCELLED = "EVENT_CANCELLED",
+  NEW_EVENT = "NEW_EVENT",
+  AFTER_EVENT = "AFTER_EVENT",
+  RESCHEDULE_EVENT = "RESCHEDULE_EVENT",
+  // Include enums from Prisma schema if they differ and are used in output
+  AFTER_HOSTS_CAL_VIDEO_NO_SHOW = "AFTER_HOSTS_CAL_VIDEO_NO_SHOW",
+  AFTER_GUESTS_CAL_VIDEO_NO_SHOW = "AFTER_GUESTS_CAL_VIDEO_NO_SHOW",
+}
+
+export enum WorkflowTimeUnit {
+  HOUR = "HOUR",
+  MINUTE = "MINUTE",
+  DAY = "DAY",
+}
+
+export enum RecipientType {
+  HOST = "HOST",
+  ATTENDEE = "ATTENDEE",
+  EMAIL = "EMAIL",
+  PHONE_NUMBER = "PHONE_NUMBER",
+}
+
+export enum StepAction {
+  EMAIL_HOST = "EMAIL_HOST",
+  EMAIL_ATTENDEE = "EMAIL_ATTENDEE",
+  EMAIL_ADDRESS = "EMAIL_ADDRESS",
+  SMS_ATTENDEE = "SMS_ATTENDEE",
+  SMS_NUMBER = "SMS_NUMBER",
+  WHATSAPP_ATTENDEE = "WHATSAPP_ATTENDEE",
+  WHATSAPP_NUMBER = "WHATSAPP_NUMBER",
+}
+
+export enum TemplateType {
+  REMINDER = "REMINDER",
+  CUSTOM = "CUSTOM",
+}
+// --- End Enums ---
+
+// --- Nested Output DTOs ---
+
+export class WorkflowMessageOutputDto {
+  @ApiProperty({
+    description: "Subject of the message",
+    example: "Reminder: Your Meeting {EVENT_NAME} - {EVENT_DATE_ddd, MMM D, YYYY h:mma} with Cal.com",
+  })
+  @Expose()
+  subject!: string;
+
+  @ApiPropertyOptional({
+    description: "HTML content of the message",
+    example: "<p>Reminder for {EVENT_NAME}.</p>",
+  })
+  @Expose()
+  html?: string;
+
+  @ApiPropertyOptional({
+    description: "Text content of the message (used for SMS/WhatsApp)",
+    example: "Reminder for {EVENT_NAME}.",
+  })
+  @Expose()
```

Comment:
The output too should follow same things i wrote for inputs:
1. Split DTOs because now our customers would have to manually check if `html` or `text` exist in the response.
2. Use types